src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -34,17 +35,28 @@ static bool simplify_gather(shared_ptr<Node> node) {
         if (data.get_partial_shape().is_dynamic() || indices.get_partial_shape().is_dynamic()) {
             return false;
         }
-        // if rank of data and gather output dont match, we will skip
-        if (data.get_shape().size() != node->get_shape().size()) {
-            return false;
-        }
 
         auto axis = gather->get_axis();
         if (axis == opset3::Gather::AXIS_NOT_SET_VALUE) {
             NGRAPH_DEBUG << "axis value not set";
             return false;
         }
 
+        if (data.get_shape().size() != node->get_shape().size()) {
+            auto constant_indices = ov::as_type_ptr<opset3::Constant>(gather->input_value(1).get_node_shared_ptr());
+            if (!constant_indices)
+                return false;
+            // case_3: if input_shape is (1,3,5,5) and axis = 0, indices = 0, then gather is just a Squeeze
+            const auto const_indices = constant_indices->cast_vector<int64_t>();
+            if (axis == 0 && data.get_shape()[axis] == 1 && const_indices.size() == 1 && const_indices[0] == 0) {
+                auto squeeze = std::make_shared<opset8::Squeeze>(gather->input_value(0), gather->input_value(2));
+                ov::replace_node(gather, squeeze);
+                return true;
+            }
+            // if rank of data and gather output don't match, we will skip
```

Comment:
The comment is confusing: we have already checked on 45th line that rank of data and gather output don't match. And we return false not because of that.
Could you please correct it or just remove? Thanks